test/test_dss_swagger_spec.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import unittest
+import requests
+from hca.util.compat import USING_PYTHON2
+
+import hca.dss
+
+if USING_PYTHON2:
+    import mock
+    from mock import mock_open
+else:
+    from unittest import mock
+    from unittest.mock import mock_open
+
+
+class TestDssSwaggerSpec(unittest.TestCase):
+    client = hca.dss.DSSClient()
+    swagger_url = ""
+    dummy_response = None
+    open_fn_name = "__builtin__.open" if USING_PYTHON2 else "builtins.open"
+
+    def setUp(self):
+        self.client._swagger_spec = None
+        self.client._spec_valid_for_days = 1
+        self.swagger_url = "test_swagger_url"
+        self.client._session = requests.Session()
+        self.client.config.__dict__.update({
+            'DSSClient': {
+                'swagger_url': self.swagger_url
+            },
+            'swagger_filename': "test_filename"
+        })
+        self.dummy_response = requests.models.Response()
+        self.dummy_response._content = b'{"swagger": ""}'
+        self.dummy_response.status_code = 200
+
+    def test_get_swagger_spec_new(self):
+        with mock.patch('os.path.exists') as mock_exists:
```

Comment:
I think this form might be more readable:
```python
with x as y, \
    a as b, \
    c as d:
    ...
```

Reply:
If you're already overriding the location of the filename, maybe you don't need to mock things like os.makedirs?

Reply:
Using `os.makedirs` mock to verify function calls. Mocks for `os.path.exists` and `load_swagger_json` could be replaced by creating temp dirs/files, however, I would opt in favor of using mocks so as to avoid littering temp dirs/files. Plus, your suggestion for readability helps a lot.